### PR TITLE
Spectral Thief should set boosts to zero on target despite abilities

### PR DESCRIPTION
**Pull request by Slayer95**
Merge only after https://github.com/Zarel/Pokemon-Showdown-Client/commit/c280c60671be767fe138a31026c3e0dbf13e4b78 is live

### CODE REVIEW

**Comment by Zarel** [comment]
Why not just use `-copyboost` and `-clearboost`?

File: data/scripts.js [hunk 1]
```diff
@@ -363,14 +363,15 @@ exports.BattleScripts = {
...
 			}
-			this.boost(boosts, target);
+			this.boost(boosts, pokemon);
 
 			for (let statName in boosts) {
-				boosts[statName] = -boosts[statName];
+				boosts[statName] = 0;
 			}
-			this.boost(boosts, pokemon);
+			target.setBoost(boosts);
```
Comment (Zarel) [inline data/scripts.js]: umm, `target.clearBoosts()` exists

File: data/scripts.js [hunk 2]
```diff
@@ -363,14 +363,15 @@ exports.BattleScripts = {
 			let boosts = {};
 			for (let statName in target.boosts) {
 				let stage = target.boosts[statName];
-				if (stage > 0) boosts[statName] = -stage;
+				if (stage > 0) boosts[statName] = stage;
 			}
-			this.boost(boosts, target);
+			this.boost(boosts, pokemon);
```
Comment (Zarel) [inline data/scripts.js]: Is this really `boost` and not `setBoost`? Does it get doubled by Simple, etc?

File: data/scripts.js (hunk 1 above)
Comment (Zarel) [inline data/scripts.js]: Unless this only clears positive boosts?

Comment (Slayer95) [inline data/scripts.js]: Only positive boosts, yes

**Comment by Slayer95** [comment]
@Zarel , it's not ``-copyboost``, because the boost stages are added, rather than cloned

**Comment by Zarel** [comment]
Does it say which boosts are added and how?

**Comment by Zarel** [comment]
I still don't like `-stealboost` as a name. Can you think of something that doesn't imply more than just that the positive boosts are cleared?

**Pull request merged**